### PR TITLE
Adding new attribute to control save the user groups origin

### PR DESCRIPTION
**Pull request by jcabrerizo**
Adds a new attribute in the WebEntitlementContext sincronized with the user session to keep the origin of the user groups

### CODE REVIEW

**Comment by algairim** [comment]
I would like to see some tests!

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/security/provider/LdapSecurityProvider.java [hunk 1]
```diff
@@ -69,6 +70,7 @@
     public static final Logger LOG = LoggerFactory.getLogger(LdapSecurityProvider.class);
 
     public static final String LDAP_CONTEXT_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+    public static final String LDAP_USER_GROUPS_ORIGIN = "brooklyn.entitlements.user.groups.origin.LdapSecurityProvider";
```
Comment (ahgittin) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/security/provider/LdapSecurityProvider.java:73]: i'd make this be `LdapSecurityProvider.class.getName()` -- there is no need for origin in the value here (it is used as a value of a key which says `origin`)

**Comment by ahgittin** [comment]
one trivial comment, ignore or address as you like, then please merge

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/security/provider/LdapSecurityProvider.java (hunk 1 above)
Comment (jcabrerizo) [reply rest/rest-resources/src/main/java/org/apache/brooklyn/rest/security/provider/LdapSecurityProvider.java]: Addressed, thanks for the feed back

**Pull request merged**